### PR TITLE
Add moves record and engine evaluation

### PR DESCRIPTION
**Pull request by KGolemo**

### CODE REVIEW

File: chess-game/board.py [hunk 1]
```diff
@@ -55,22 +65,38 @@ def __init__(self, parent):
...
             col, row = self.get_square_coords(sqr_index)
-            piece = self.board.piece_at(sqr_index)
+            piece: PieceItem = self.board.piece_at(sqr_index)
             if piece:
                 piece_label = PieceItem(self, piece)
                 self.layout.addWidget(piece_label, row, col)
                 self.pieces_items[sqr_index] = piece_label
 
+    def update_pieces(self, next_board):
+        for sqr_index in range(64):
```
Comment (rozchlastywacz) [inline chess-game/board.py]: magic number: 64

File: chess-game/board.py [hunk 2]
```diff
@@ -133,76 +161,49 @@ def draw_possible_moves(self, square_index):
                 self.set_square_style(sq_idx, "frame")
 
     def move_piece(self, square_index):
-        self.get_possible_moves()
         if (
             self.previous_sq_idx in self.possible_moves.keys()
             and square_index in self.possible_moves[self.previous_sq_idx]
         ):
+            next_board = self.board.copy()
```
Comment (rozchlastywacz) [inline chess-game/board.py]: why do you save every other variable to object field, but not this one?

File: chess-game/info.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+import chess.engine
+from itertools import pairwise
+
+
+class MyQTableWidget(QTableWidget):
+    def __init__(self):
+        super().__init__()
+
+    def wheelEvent(self, event):
+        delta = event.angleDelta().y() // 120
```
Comment (rozchlastywacz) [inline chess-game/info.py:24]: magic number: 120

File: chess-game/info.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    def update_engine_evaluation(self, evaluation):
+        score = evaluation[0]["score"].white()
+        if score.score() is not None:
+            self.centipawns = score.score()
+            self.mate = None
+        else:
+            self.centipawns = None
+            self.mate = score
+        self.update()
+
```
Comment (rozchlastywacz) [inline chess-game/info.py:161]: legacy code?

File: chess-game/info.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+                )
+            else:
+                mapped_line = [
+                    f"{m1.uci()} {m2.uci()}" for m1, m2 in pairwise(line[2:6])
+                ]
+                line_str += str(" ") + line[1].uci()
+                line_str += "".join(
+                    f" {j+1}. {m}"
+                    for j, m in enumerate(mapped_line, move_index // 2 + 1)
+                )
```
Comment (rozchlastywacz) [inline chess-game/info.py]: could be separate method to comment itself

File: chess-game/main.py [hunk 6]
```diff
@@ -1,18 +1,23 @@
...
 
 class MainWindow(QMainWindow):
 
     def __init__(self):
         super(MainWindow, self).__init__()
 
+        engine = chess.engine.SimpleEngine.popen_uci(
+            "stockfish/stockfish-windows-x86-64-avx2.exe"
+        )
+
```
Comment (rozchlastywacz) [inline chess-game/main.py:17]: use env variable or something similar

File: chess-game/utils.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import math
+
+
+def sigmoid(x):
+    return 1 / (1 + math.exp(-0.3 * x))
```
Comment (rozchlastywacz) [inline chess-game/utils.py:5]: nice

**Review by rozchlastywacz — APPROVED** [review]
lgtm with some exceptions that could be fixed later
Awch

**Review by layrell — APPROVED** [review]

**Pull request merged**